src/application.rs

```diff
@@ -59,13 +64,30 @@ impl Application {
             ui,
             ipc_tx: None,
             model,
-            raw_model: RawModel::new(),
+            pending_requests,
         })
     }
-
-    pub fn send_ipc_message(&self, msg: IpcMessage) {
+    pub fn send_ipc_message<F>(&mut self, msg: IpcMessage, handle_response: F)
+    where
+        F: Fn(&mut Application) -> () + 'static,
+    {
         if let Some(ipc_tx) = &self.ipc_tx {
-            ipc_tx.send(msg).unwrap();
+            match &msg {
+                IpcMessage::Request { request, id } => {
+                    debug!("Pending response for: {:?}", request);
+                    self.pending_requests.insert(*id, Rc::new(handle_response));
+                }
+                _ => {}
```

Comment:
Do you want to send any message here through the channel or just Request one? Or you just want to put Requests inside pending_requests? If former then you need to print error and return here, if latter -- you might want to use 

```rust
if let IpcMessage::Request(request, id) = msg {
  ...
}
```